device.cpp

```diff
@@ -0,0 +1,25 @@
+#include "device.h"
+
+// Constructor
+Device::Device(const std::string& type, const std::string& deviceId, bool status, const std::vector<Sensor> sensors)
```

Comment:
when you send std::vector<> (or any large objects) as parameter, you should send it as a const reference. This way you will avoid an unnecessary copy (if it's not clear, quick search value vs cons reference for examples and explaining). Apply this for your entire project.